src/typing/type-inferencer.ts

```diff
@@ -0,0 +1,35 @@
+import { DefaultNormalizedAstFold } from '../abstract-interpretation/normalized-ast-fold';
+import type { RLogical } from '../r-bridge/lang-4.x/ast/model/nodes/r-logical';
+import type { RNumber } from '../r-bridge/lang-4.x/ast/model/nodes/r-number';
+import type { RString } from '../r-bridge/lang-4.x/ast/model/nodes/r-string';
+
+export enum RDataType {
+    Logical = 'LogicalType',
+    Numeric = 'NumericType',
+    String = 'StringType',
+}
+
+export type TypingInfo = RDataType | undefined;
```

Comment:
What is the intended Semantic of `undefined`? Is it to be equivalent to `any` or should it just signal that we have no type information? 
(I mean given an incremental typing approach, TOP may be equialvent to "no information" but we may want to separate "we know that we know nothing" from "we haven't tried (yet)")

So if this should be TOP/any I would make it explicit, otherwise I would at least clarify the meaning of `undefined` with a comment.